### PR TITLE
Pretty Difference. 

### PR DESCRIPTION
**Pull request by shubham-cmyk**

### CODE REVIEW

File: pkg/runner/operations/assert.go [hunk 1]
```diff
@@ -2,29 +2,78 @@ package operations
...
+	candidateContentBytes, _ := yaml.Marshal(candidateContent)
+
+	diff := difflib.UnifiedDiff{
+		A:        difflib.SplitLines(string(expectedContentBytes)),
+		B:        difflib.SplitLines(string(candidateContentBytes)),
+		FromFile: "Expected",
+		ToFile:   "Actual",
+		Context:  3,
+	}
+	diffStr, _ := difflib.GetUnifiedDiffString(diff)
```
Comment (eddycharly) [inline pkg/runner/operations/assert.go:77]: checking errors is not optional

File: pkg/runner/operations/assert.go [hunk 2]
```diff
@@ -2,29 +2,78 @@ package operations
...
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 func Assert(ctx context.Context, expected unstructured.Unstructured, c client.Client) error {
-	return wait.PollUntilContextCancel(ctx, interval, false, func(ctx context.Context) (bool, error) {
-		if candidates, err := read(ctx, expected, c); err != nil {
+	var differences []string
+	var successfulMatches int
+	var totalCandidatesChecked int
+	var candidates []unstructured.Unstructured
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply pkg/runner/operations/assert.go:21]: you don't want all the changes, only the last state just before the timeout occurs

[the remark this answers is not in the document]
Comment (shubham-cmyk) [reply pkg/runner/operations/assert.go:21]: so you mean only 1  difference is fine ?

[the remark this answers is not in the document]
Comment (eddycharly) [reply pkg/runner/operations/assert.go:21]: not a single difference, but the differences detected in the last run.

**Review by eddycharly — APPROVED** [review]

**Pull request merged**